cli/src/commit_templater.rs

```diff
@@ -497,6 +498,17 @@ impl<'repo> CoreTemplatePropertyVar<'repo> for CommitTemplatePropertyKind<'repo>
         }
     }
 
+    fn try_into_string_pattern(
+        self,
+    ) -> Option<BoxedTemplateProperty<'repo, jj_lib::str_util::StringPattern>> {
```

Comment:
I don't think we'll need support for dynamic construction of string patterns. See callers of `template_parser::expect_string_literal()` for parsing of static values.

Reply:
> we could write something like `expect_string_literal`, but that's not going to work in the existing `contains` function which tries to stringify;

That's another reason to not implement `contains(pattern)`.

BTW, in order to support arbitrary `kind:(expr ++ ..)` pattern, we can parse the `kind:` part statically, and evaluate `(expr ++ ..)` as stringifiable expression. I personally wouldn't try this route.